src/main/java/org/opensrp/web/rest/StockResource.java

```diff
@@ -203,4 +218,111 @@ public Stock update(Stock stock) {
+			return new ResponseEntity<>(HttpStatus.CREATED);
+		}
+	}
+
+	@DeleteMapping(value = "/{id}", produces = { MediaType.APPLICATION_JSON_VALUE })
+	public ResponseEntity<String> delete(@PathVariable(Constants.RestPartVariables.ID) Long id) {
+		if (id == null) {
+			return new ResponseEntity<>("Stock item id is required", RestUtils.getJSONUTF8Headers(),
+					HttpStatus.BAD_REQUEST);
+		} else {
+			stockService.deleteStock(id);
+			return new ResponseEntity<>("Stock deleted successfully", RestUtils.getJSONUTF8Headers(),
+					HttpStatus.NO_CONTENT);
+		}
+	}
+
+	@GetMapping(value = "/servicePointId/{servicePointId}", produces = {
+			MediaType.APPLICATION_JSON_VALUE })
+	public ResponseEntity<List<Stock>> getStockItemsByServicePoint(@PathVariable String servicePointId) {
+		return new ResponseEntity<>(stockService.getStocksByServicePointId(servicePointId),
+				RestUtils.getJSONUTF8Headers(),
+				HttpStatus.OK);
+	}
+
+	@PostMapping(headers = { "Accept=multipart/form-data" }, produces = {
+			MediaType.APPLICATION_JSON_VALUE }, value = "/import/inventory")
+	public ResponseEntity importInventoryData(@RequestParam("file") MultipartFile file)
+			throws IOException {
+		Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+		String userName = authentication.getName();
+		List<Map<String, String>> csvClients = readCSVFile(file);
+		CsvBulkImportDataSummary csvBulkImportDataSummary = stockService
+				.convertandPersistInventorydata(csvClients, userName);
+
+		generateCSV(csvBulkImportDataSummary);
+		File csvFile = new File(SAMPLE_CSV_FILE);
+
+		if (csvBulkImportDataSummary != null && csvBulkImportDataSummary.getFailedRecordSummaryList().size() > 0) {
+			return ResponseEntity.badRequest()
+					.header("Content-Disposition", "attachment; filename=" + "importsummaryreport" + ".csv")
+					.contentLength(csvFile.length())
+					.contentType(MediaType.parseMediaType("text/csv"))
+					.body(new FileSystemResource(csvFile));
+		} else {
+			return ResponseEntity.ok()
+					.header("Content-Disposition", "attachment; filename=" + "importsummaryreport" + ".csv")
+					.contentLength(csvFile.length())
+					.contentType(MediaType.parseMediaType("text/csv"))
+					.body(new FileSystemResource(csvFile));
+		}
+	}
+
+	private List<Map<String, String>> readCSVFile(MultipartFile file) throws IOException {
+		List<Map<String, String>> csvClients = new ArrayList<>();
+		try (Reader reader = new InputStreamReader(file.getInputStream());
+		     CSVParser parser = new CSVParser(reader, CSVFormat.DEFAULT.withHeader())) {
+
+			List<CSVRecord> records = parser.getRecords();
+			for (CSVRecord record : records) {
+				csvClients.add(record.toMap());
+			}
+		}
+		return csvClients;
+	}
+
+	private void generateCSV(CsvBulkImportDataSummary csvBulkImportDataSummary) {
+		try (
+				BufferedWriter writer = Files.newBufferedWriter(Paths.get(SAMPLE_CSV_FILE));
+				CSVPrinter csvPrinter = new CSVPrinter(writer, CSVFormat.DEFAULT);
```

Comment:
Always generate a new CSV every time so that there is conflict of sessions when users are uploading concurrenlly

Reply:
Done